upup/models/cloudup/resources/addons/karpenter.sh/k8s-1.19.yaml.template

```diff
@@ -639,286 +694,295 @@ data:
+    app.kubernetes.io/managed-by: Helm
 rules:
-- apiGroups: [""]
-  resources: ["namespaces"]
-  verbs: ["get", "list", "watch"]
-- apiGroups: [""]
-  resources: ["secrets"]
-  verbs: ["get", "list", "watch", "update"]
-- apiGroups: ["coordination.k8s.io"]
-  resources: ["leases"]
-  verbs: ["get", "watch", "create", "update"]
+  # Read
+  - apiGroups: [""]
+    resources: ["services"]
+    resourceNames: ["kube-dns"]
+    verbs: ["get"]
 ---
-# Source: karpenter/templates/controller/rbac.yaml
+# Source: karpenter/templates/rolebinding.yaml
 apiVersion: rbac.authorization.k8s.io/v1
 kind: RoleBinding
 metadata:
-  name: karpenter-controller
+  name: karpenter
   namespace: kube-system
+  labels:
+    helm.sh/chart: karpenter-v0.24.0
+    app.kubernetes.io/name: karpenter
+    app.kubernetes.io/instance: karpenter
+    app.kubernetes.io/version: "0.24.0"
+    app.kubernetes.io/managed-by: Helm
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: Role
-  name: karpenter-controller
-subjects:
-- kind: ServiceAccount
   name: karpenter
-  namespace: kube-system
+subjects:
+  - kind: ServiceAccount
+    name: karpenter
+    namespace: kube-system
 ---
-# Source: karpenter/templates/webhook/rbac.yaml
+# Source: karpenter/templates/rolebinding.yaml
 apiVersion: rbac.authorization.k8s.io/v1
 kind: RoleBinding
 metadata:
-  name: karpenter-webhook
+  name: karpenter-dns
   namespace: kube-system
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: Role
-  name: karpenter-webhook
+  name: karpenter-dns
 subjects:
-- kind: ServiceAccount
-  name: karpenter
-  namespace: kube-system
+  - kind: ServiceAccount
+    name: karpenter
+    namespace: kube-system
 ---
-# Source: karpenter/templates/controller/deployment.yaml
+# Source: karpenter/templates/service.yaml
 apiVersion: v1
 kind: Service
 metadata:
-  name: karpenter-metrics
+  name: karpenter
   namespace: kube-system
 spec:
+  type: ClusterIP
   ports:
-    - port: 8080
+    - name: http-metrics
+      port: 8080
       targetPort: http-metrics
-  selector:
-    karpenter: controller
----
-# Source: karpenter/templates/webhook/deployment.yaml
-apiVersion: v1
-kind: Service
-metadata:
-  name: karpenter-webhook
-  namespace: kube-system
-spec:
-  ports:
-    - port: 443
+      protocol: TCP
+    - name: https-webhook
+      port: 443
       targetPort: https-webhook
+      protocol: TCP
   selector:
-    karpenter: webhook
+    app.kubernetes.io/name: karpenter
+    app.kubernetes.io/instance: karpenter
+
+
+
 ---
-# Source: karpenter/templates/controller/deployment.yaml
+# Source: karpenter/templates/deployment.yaml
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: karpenter-controller
+  name: karpenter
   namespace: kube-system
 spec:
   replicas: {{ ControlPlaneControllerReplicas false }}
+  revisionHistoryLimit: 10
   strategy:
-    type: Recreate
+    rollingUpdate:
+      maxUnavailable: 1
   selector:
     matchLabels:
-      karpenter: controller
```

Comment:
This will break upgrades I think. Selectors are immutable.

Reply:
Good catch, I have changed it to the old value
 - https://github.com/kubernetes/kops/pull/15144/commits/28da422b8429990d6117d9b76bdc1007e21c164b

Reply:
Maybe we can switch to a new deployment named "karpenter" without controller or webhook in the name because now one pods handle both.